cmd/soroban-rpc/internal/daemon/daemon.go

```diff
@@ -299,28 +300,44 @@ func (d *Daemon) mustInitializeStorage(cfg *config.Config) *feewindow.FeeWindows
 	defer cancelReadTxMeta()
 	var initialSeq uint32
 	var currentSeq uint32
-	dataMigrations, err := db.BuildMigrations(readTxMetaCtx, d.logger, d.db, cfg)
+	applicableRange, err := db.GetMigrationLedgerRange(readTxMetaCtx, d.db, cfg.HistoryRetentionWindow)
 	if err != nil {
-		d.logger.WithError(err).Fatal("could not build migrations")
+		d.logger.WithError(err).Fatal("could not get ledger range for migration")
 	}
 
-	// Merge migrations range and fee stats range to get the applicable range
-	latestLedger, err := db.NewLedgerEntryReader(d.db).GetLatestLedgerSequence(readTxMetaCtx)
+	maxFeeRetentionWindow := max(cfg.ClassicFeeStatsLedgerRetentionWindow, cfg.SorobanFeeStatsLedgerRetentionWindow)
```

Comment:
do you know why the soroban fee stats retention window is different from the history retention window?

Reply:
I thought it was a part of requirement or something that we follow in Horizon for calculating fee stats to consider last 50 ledgers. @2opremio  would know it better tho.

Reply:
It’s because the soroban activity is much lower and a larger window will increase the chances of including actual soroban transactions. On the other hand we want to the metric to contain recent information so we want to include a short recent window (hence why it’s different and shorter for classic)